### DISCUSSION TITLE
How to use Render Mode

### DISCUSSION THREAD

**Discussion by maggiduscher** [roflmuffin/CounterStrikeSharp#437] [Q&A] [answered]
I know that in CSGO/Sourcemod you could use SetEntityRenderColor and SetEntityRenderMode to Color the PlayerModel.

I assume in CSS the same would be player.PlayerPawn.Value.Render = Color(255,0,0) and e.g. player.PlayerPawn.Value.RenderMode = RenderMode_t.kRenderTransColor

i tried to color the model in combination with every RenderMode (in MM/Sourcemod i used TransColor) but can't seem to get it to work.

Does anybody know, what i am doing wrong?

**Comment by roflmuffin** [comment, marked as answer]
Are you calling the set state changed method (which marks the field as needing update to clients)

`player.PlayerPawn.Value.Render = Color.FromArgb(255, 255, 255, 255);`
`Utilities.SetStateChanged(player.PlayerPawn.Value, "CBaseModelEntity", "m_clrRender");`

**Reply by maggiduscher** [reply]
Adding "SetStateChanged" worked, thanks. But how do i know which changes need to be actively triggered via "SetStateChanged". Is there a underlying documentation i need to take a look at? Cause i can't seem to find the link between "SetStateChanged" and "player.PlayerPawn.Value.Render" in the CSSharp Documentation

**Reply by roflmuffin** [reply]
It's hard to know, it's mostly just guess work based on how valve does networking. Like health for instance, the client would only see the new health the next time their health is changed (take damage) or unless you state change.
